google_guest_agent/run/run.go

```diff
@@ -0,0 +1,97 @@
+//  Copyright 2023 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// Package run is a package with utilities for running command and handling results.
+package run
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+// Result wraps a command execution result.
+type Result struct {
+	// Exit code. Set to -1 if we failed to run the command.
+	ExitCode int
+	// Stderr or err.Error if we failed to run the command.
+	StdErr string
+	// Stdout or "" if we failed to run the command.
+	StdOut string
+}
+
+// Error return an error containing the stderr content.
+func (e Result) Error() string {
+	return strings.TrimSuffix(e.StdErr, "\n")
+}
+
+// Quiet runs a command and doesn't return any status.
```

Comment:
is this comment accurate? we're returning `res` on error right?

Reply:
Ah! I meant to change this comment, it was wrong before me, it meant it doesn't return status but returns error. I'll get it changed. Thanks.

Reply:
Done. Thanks.